web/src/components/tasks/TaxInput.tsx

```diff
@@ -126,11 +143,14 @@ export const TaxInput = (props: Props): ReactElement => {
               </Alert>
             ) : (
               <>
-                <OnboardingTaxId required />
+                <OnboardingTaxId
+                  required={isAuthenticated === IsAuthenticated.FALSE ? false : true}
+                  handleChangeOverride={showNeedsAccountModalForGuest()}
```

Comment:
So this is so we don't have inline errors when the user tries to enter data here (I think ideally I would like to relocate this logic one level lower but for some reason that isn't working at all)

Reply:
JK that doesn't work at all because its spreading props into its children so it needs to not be required to not be validated in addition to modifying the validation function itself and that doesn't feel particularly less complicated